### PR TITLE
Move prime functions

### PR DESCRIPTION
**Pull request by goldenMetteyya**

### CODE REVIEW

**Comment by goldenMetteyya** [comment]
@dignifiedquire Hey I moved the prime related functions behind a features tag and also changed the benchmarks to use criterion.

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [package]
-authors = ["The Rust Project Developers"]
+authors = ["dignifiedquire <dignifiedquire@gmail.com>", "organism <hello@organism.ai>"]
```
Comment (dignifiedquire) [inline Cargo.toml]: please keep the original authors in, they wrote a lot of this code

File: src/algorithms.rs [hunk 2]
```diff
@@ -731,4 +870,138 @@ mod algorithm_tests {
...
+
+
+
+    #[test]
+    fn test_mod_inverse() {
+        let tests = [
+            ["1234567", "458948883992"],
+	    ["239487239847", "2410312426921032588552076022197566074856950548502459942654116941958108831682612228890093858261341614673227141477904012196503648957050582631942730706805009223062734745341073406696246014589361659774041027169249453200378729434170325843778659198143763193776859869524088940195577346119843545301547043747207749969763750084308926339295559968882457872412993810129130294592999947926365264059284647209730384947211681434464714438488520940127459844288859336526896320919633919"],
+	    ["-10", "13"],
+            ["-6193420858199668535", "2881"],
```
Comment (dignifiedquire) [inline src/algorithms.rs]: formatting seems off, maybe just run `cargo fmt` on all the files

Comment (goldenMetteyya) [reply src/algorithms.rs]: Okie done

**Comment by dignifiedquire** [comment]
Thank you

**Pull request merged**